lib/sqlmap/sqlmap_manager.rb

```diff
@@ -8,43 +8,56 @@ def initialize(session)
 
     def new_task
       res = @session.get('/task/new')
-      return JSON.parse(res.body)
+      parse_response(res)
     end
 
     def delete_task(task_id)
       res = @session.get('/task/' + task_id + '/delete')
-      return JSON.parse(res.body)
+      parse_response(res)
     end
 
     def set_option(task_id, key, value)
       post = { key => value }
       res = @session.post('/option/' + task_id + '/set', nil, post.to_json, {'ctype' => 'application/json'})
-      return JSON.parse(res.body)
+      parse_response(res)
     end
 
     def get_options(task_id)
       res = @session.get('/option/' + task_id + '/list')
-      return JSON.parse(res.body)
+      parse_response(res)
     end
 
     def start_task(task_id, options = {})
       res = @session.post('/scan/' + task_id + '/start' , nil, options.to_json, {'ctype' => 'application/json'})
-      return JSON.parse(res.body)
+      parse_response(res)
+
     end
 
     def get_task_status(task_id)
       res = @session.get('/scan/' + task_id + '/status')
-      return JSON.parse(res.body)
+      parse_response(res)
     end
 
     def get_task_log(task_id)
       res = @session.get('/scan/' + task_id + '/log')
-      return JSON.parse(res.body)
+      parse_response(res)
     end
 
     def get_task_data(task_id)
       res = @session.get('/scan/' + task_id + '/data')
-      return JSON.parse(res.body)
+      parse_response(res)
+    end
+
+    private
+    def parse_response(res)
+      if res && res.body
+        begin
+          res = JSON.parse(res.body)
+        rescue JSON::ParserError
+        end
+      end
+
+      res
```

Comment:
@wchen-r7 That is true. The code added in plugins/sqlmap.rb is:
```ruby
taskid = @manager.new_task
if taskid && taskid['taskid']
  taskid = taskid['taskid']
  @hid_tasks[(@hid_tasks.length + 1).to_s] = taskid
  task_options = @manager.get_options(taskid)
  @tasks[@hid_tasks[@hid_tasks.length]] = task_options['options']
  print_good("Created task: #{@hid_tasks.length}")
else
  print_error("Error connecting to the server. Please make sure the sqlmapapi server is running at #{@host}:#{@port}")
end
```
If it is a hash, the taskid will be extracted. If it is a null, it is will be handled by the if statement. Are we good to go here? If yes then I am going to implement checks in the other API calls as well.